From b59dff5178712b4eefafd02bf40837fe9d11b624 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Fri, 11 Oct 2013 12:09:37 +0200
Subject: wined3d: Invoke surface_unload through the CS in
 wined3d_surface_update_desc.

This fixes another case where we use a GL context in the main thread and
crash fglrx in doing so.
---
 dlls/wined3d/texture.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dlls/wined3d/texture.c b/dlls/wined3d/texture.c
index 765192d..f42127e 100644
--- a/dlls/wined3d/texture.c
+++ b/dlls/wined3d/texture.c
@@ -1033,7 +1033,10 @@ HRESULT CDECL wined3d_texture_update_desc(struct wined3d_texture *texture, UINT
     }
 
     if (device->d3d_initialized)
-        texture->resource.resource_ops->resource_unload(&texture->resource);
+    {
+        wined3d_cs_emit_evict_resource(device->cs, &texture->resource);
+        device->cs->ops->finish(device->cs);
+    }
 
     sub_resource = &texture->sub_resources[0];
     surface = sub_resource->u.surface;
-- 
2.8.0

